### ISSUE TITLE
Loading songs from Uri leaks memory

### ISSUE THREAD

**Issue by NaamloosDT** [MonoGame/MonoGame#5909]
I'm not completely sure whether this is causes by Monogame, but the mediaplayer seems to increase RAM usage each time a new song is loaded. Any ideas?

**Comment by KonajuGames** [comment]
Which platform? The media player implementation is different for different platforms.

**Comment by NaamloosDT** [comment]
Windows (DX)

**Comment by tomspilman** [comment]
@NaamloosDT -  I assume you are doing something like this:

```
var song = Song.FromUri("Song Name");
MediaPlayer.Play(song);
```

Are you later disposing the `Song` object?

**Comment by NaamloosDT** [comment]
No, Though I'm not loading them over each time, I'm only loading them into memory once and storing them

**Comment by tomspilman** [comment]
> I'm only loading them into memory once and storing them

Then it does sound like there could be some sort of issue.

**Comment by NaamloosDT** [comment]
Hmm, weird.. let me check whether anything is wrong from my side. Will report back! :)

**Comment by NaamloosDT** [comment]
My local copy of my game is disposing Song objects and loading new one, yet still seems to increase RAM usage..

**Comment by NaamloosDT** [comment]
Taskmanager tells me it's about 1MB per map change

**Comment by tomspilman** [comment]
> yet still seems to increase RAM usage..

Thanks for double checking this @NaamloosDT !

**Comment by NaamloosDT** [comment]
No problem. I've tried trying to find a fix earlier but I'm not too familiar with monogame's code :s

**Comment by FredrikRosenqvist** [comment]
Hello there, I was thinking about contributing to Monogame and this issue seems like a good start and something I can look into. @NaamloosDT Do you have a scenario I can setup where I can replicate the memory leak?

**Comment by NaamloosDT** [comment]
Not really, the only code I have available is very case-specific.
You could take a look at Rizumu (my game) but I don't have any maps ready for you to try it with (you'd need multiple in this case).
